src/Suave/Cookie.fs

```diff
@@ -42,211 +42,224 @@ module Cookie =
+  let refreshCookies relativeExpiry httpCookie : WebPart =
+    slidingExpiry relativeExpiry httpCookie ||> setPair
+
+  let updateCookies (csctx : CookiesState) f_plainText : WebPart =
+    context (fun ctx ->
+      let logger = ctx.runtime.logger
+      let plainText =
+        match readCookies csctx.serverKey csctx.cookieName ctx.response.cookies with
+        | Choice1Of2 (_, plainText) ->
+          Log.log logger "Suave.Cookie.updateCookies" LogLevel.Debug "updateCookies - existing"
+          f_plainText (Some plainText)
         | Choice2Of2 _ ->
-          Log.log logger "Suave.Cookie.update_cookies" LogLevel.Debug "update_cookies - first time"
-          f_plain_text None
+          Log.log logger "Suave.Cookie.updateCookies" LogLevel.Debug "updateCookies - first time"
+          f_plainText None
 
       /// Since the contents will completely change every write, we simply re-generate the cookie
-      generate_cookies csctx.server_key csctx.cookie_name
-                       csctx.relative_expiry csctx.secure
-                       plain_text'
-      ||> set_pair
-      >>= Writers.set_user_data csctx.user_state_key plain_text')
+      generateCookies csctx.serverKey csctx.cookieName
+                       csctx.relativeExpiry csctx.secure
+                       plainText
+      ||> setPair
+      >>= Writers.setUserData csctx.userStateKey plainText)
 
-  let cookie_state (csctx : CookiesState)
+  let cookieState (csctx : CookiesState)
                    // unit -> plain text to store OR something to run of your own!
-                   (no_cookie : unit -> Choice<byte [], WebPart>)
-                   (decryption_failure   : _ -> Choice<byte [], WebPart>)
+                   (noCookie : unit -> Choice<byte [], WebPart>)
+                   (decryptionFailure   : _ -> Choice<byte [], WebPart>)
                    (f_success : WebPart)
                    : WebPart =
     context (fun ({ runtime = { logger = logger }} as ctx) ->
 
       let log = Log.log logger "Suave.Cookie.cookie_state" LogLevel.Debug
 
-      let set_cookies plain_text =
-        let http_cookie, client_cookie =
-          generate_cookies csctx.server_key csctx.cookie_name
-                           csctx.relative_expiry csctx.secure
-                           plain_text
-        set_pair http_cookie client_cookie >>=
-          Writers.set_user_data csctx.user_state_key plain_text
+      let setCookies plainText =
+        let httpCookie, clientCookie =
+          generateCookies csctx.serverKey csctx.cookieName
+                           csctx.relativeExpiry csctx.secure
+                           plainText
+        setPair httpCookie clientCookie >>=
+          Writers.setUserData csctx.userStateKey plainText
 
-      match read_cookies csctx.server_key csctx.cookie_name (ctx.request |> HttpRequest.cookies) with
-      | Choice1Of2 (http_cookie, plain_text) ->
+      match readCookies csctx.serverKey csctx.cookieName ctx.request.cookies with
```

Comment:
Shouldn't we have some extra/other facility to make it explicit when we're incurring a parsing cost?